File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] arr, int target) {
        int n = arr.length;
        int[] ans = new int[2];

        for(int i = 0; i < n; i++) {
            for(int j = i + 1; j < n; j++) {
                if(arr[i] + arr[j] == target) {
                    ans[0] = i;
                    ans[1] = j;
                    break;
                }
            }
        }

        return ans;
    }
}

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();

        unordered_map<int, int> mp; // store val, index

        for (int i = 0; i < n; i++) {
            int remaining = target - nums[i];
            // check remaing value present in map or not.
            if (mp.find(remaining) != mp.end()) {
                return {mp[remaining], i};
            }

            mp[nums[i]] = i;
        }

        return {};
    }
};